src/statistics.jl

```diff
@@ -1109,6 +1109,21 @@ function apply_statistic(stat::BoxplotStatistic,
     nothing
 end
 
+struct CandlestickStatistic <: Gadfly.StatisticElement end
+
+"""
+    Stat.candlestick[()]
+"""
+const candlestick = CandlestickStatistic
+
+function apply_statistic(stat::CandlestickStatistic, scales, coord, aes)
+    hlgroup = aes.upper_hinge .> aes.lower_hinge
+    color_scale = get(scales, :color, Scale.color_discrete())
```

Comment:
`Scale.color_discrete` will default to Gadfly's discrete color palette.  Would users find `Scale.color_discrete_manual("white","black", levels=[true,false])` a better default?

Reply:
Well, in case of stock market:
- In Asia, a red candlestick denotes price rising.
- But in other cases, a red candlestick denotes price falling.

So, I think the Gadfly default color palette is okay, and let user set it manually.

Reply:
A potential issue here is that Gadfly assigns the first color (deepskyblue) to hlgroup[1], and hlgroup[1] could be true or false.  You can make colors and levels consistent by specifying levels, so in L1121 change `Scale.color_discrete()` to e.g. `Scale.color_discrete(levels=[true, false])`. You can choose whether deepskyblue means true or false wrt `hlgroup`.